quickwit-backward-compat/build.rs

```diff
@@ -17,15 +17,58 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program. If not, see <http://www.gnu.org/licenses/>.
 
-use std::collections::BTreeSet;
+use std::collections::{BTreeSet, HashMap};
 
 use byte_unit::Byte;
 use quickwit_config::{
     DocMapping, IndexingResources, IndexingSettings, MergePolicy, SearchSettings, SourceConfig,
 };
 use quickwit_index_config::SortOrder;
 use quickwit_metastore::checkpoint::{Checkpoint, CheckpointDelta, PartitionId, Position};
-use quickwit_metastore::{IndexMetadata, SplitMetadata};
+use quickwit_metastore::{Index, IndexMetadata, Split, SplitMetadata, SplitState};
+
+fn sample_index_for_regression() -> Index {
+    let index = sample_index_metadata_for_regression();
+
+    let split_metadata = sample_split_metadata_for_regression();
+    let split = Split {
+        split_state: SplitState::Published,
+        split_metadata,
+        update_timestamp: 1789,
+    };
+    let mut splits = HashMap::new();
+    splits.insert(split.split_id().to_string(), split);
+
+    Index {
+        metadata: index,
+        splits,
+        discarded: false,
+    }
+}
+
+fn save_index_test_files() -> anyhow::Result<()> {
```

Comment:
I think we need to have `file_backed` or something in that kind of method name. It is very hard for the reader to undertsand the different between this thing and the index_metadat thing